COORDINATE_SYSTEM2/COORDINATE_SYSTEM2/CoordinateSystem.cs

```diff
@@ -6,21 +6,89 @@
 
 namespace COORDINATE_SYSTEM2
 {
+    /// <summary>
+    /// Клас Координатна система.
+    /// Містить Ліст всіх створених точок та Ліст всіх знайдених дистанцій.
+    /// Містить Add(), який створює нову точку, добавляє її в Ліст, шукає відстані 
+    /// між існуючими точками до нової точки і добавляє їх в ліст.
+    /// Містить DisplayAllPoints(), який друкує всі точки з ліста.
+    /// Містить MaxDistance() для пошуку максимальної відстані між точками.
+    /// Містить PrintAllDistances(), який друкує всі наявні відстані з Ліста.
+    /// </summary>
     internal class CoordinateSystem
     {
         List<Point> listOfPoints = new List<Point>();
 
+        List<Rectangle> listOfRectangles = new List<Rectangle>();
+
+        List<Triangle> listOfTriangles = new List<Triangle>();
+
+        List<IFigure> listOfFigures = new List<IFigure>();
+
         List<PointsDistance> pointsDistances = new List<PointsDistance>();
 
-        public void Add(Point point)
+        //TODO мені потрібно зробити КЛАС!!! "ФІГУРА"
+        //- який буде базовим для Точки, Прямокутника та Трикутника.
+        //І в методі Ед приймати їх через цей базовий клас.
+        public void Add(IFigure figure) 
+        {
+            if(figure is Point)
+            {
+                foreach (Point point1 in listOfPoints)
+                {
+                    PointsDistance distance = new PointsDistance((Point)figure, point1);
+                    pointsDistances.Add(distance);
+                }
+
+                listOfPoints.Add((Point)figure);
+            }
+            //TODO якщо це чотирикутник.
+            //я повинен пройтися по кожній з його точок почергово і знайти відстані між уже існуючими точками до кожної нової точки.
+            //цикл по точках прямокутника, який братиме першу точку прямокутника і тоді запускатиме з нею цикл по всіх уже існучих точках.
+            else if(figure is Rectangle)
+            {
+                listOfRectangles.Add((Rectangle)figure);
+            }
+            else if(figure is Triangle)
+            {
+                listOfTriangles.Add((Triangle)figure);
+            }
+        }
+
+        public void PrintAreas(IFigure figure)
         {
-            foreach (Point point1 in listOfPoints)
+            if (figure is Point)
+            {
+                Console.WriteLine("N/A");
+            }
+            else if (figure is Rectangle)
             {
-                PointsDistance distance = new PointsDistance(point, point1);
-                pointsDistances.Add(distance);
+                Console.WriteLine(figure.ToString());
+                Console.WriteLine($"Area is {figure.Area()}");
             }
+            else if (figure is Triangle)
+            {
+                Console.WriteLine(figure.ToString());
+                Console.WriteLine($"Area is {figure.Area()}");
+            }
```

Comment:
as you can see there is no difference between Triangle and Rectangle - you calling same methods, so it's basically the same code

Reply:
you just have to work on uniting it with points, you can just print "N/A" if `Area()` returns 0 right?

Reply:
same for PrintPerimeters method